### ISSUE TITLE
PostHog Development on Apple Silicon

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#2916]
All of this will probably work with Rosetta 2, but to get it running natively on ARM, here's the situation.

1. [Homebrew](https://brew.sh/) [works well with ARM](https://github.com/Homebrew/brew/issues/10152) now. Some packages are broken, but most work.
2. Homebrew can be used to install a native version python 3.9 (`brew install python@3.9`)
3. NodeJS v15 works natively on ARM if installed via [nvm](https://github.com/nvm-sh/nvm) (`nvm use v15`)
4. ~~For postgres I still use the intel-based [postgres.app](https://postgresapp.com/)~~
[2020-01-24] Intel based postgres.app doesn't work because it share some library with psycopg2. Install via `brew install postgresql`
5. ~~Redis via homebrew should work (didn't verify as I still have an old install via macports)~~
[2020-01-24] `brew install redis`
6. Docker (Clickhouse, etc) --> not yet. There's a [tech preview](https://www.docker.com/blog/download-and-try-the-tech-preview-of-docker-desktop-for-m1/) out, but I haven't tried it yet. It's probably still broken.
7. Not all python packages install. Here's a list of holdouts:
- `grpcio` - 1.34.0 doesn't work yet - [track here](https://github.com/grpc/grpc/issues/25082)
- `pandas` - 1.2.0 doesn't work yet
- `numpy` - 1.19.5 doesn't work yet
- `cffi` - 1.14.0 fails, but 1.14.4 works, just needs to be bumped

It seems like `grpcio` can be patched to work and there's an issue for [removing numpy and pandas](https://github.com/PostHog/posthog/issues/2248), so technically with a bit of work it should be possible to develop posthog on M1 macs natively. 

Working on EE/Clickhouse is another question and will need to wait until the Apple Silicon Docker matures. Considering how much money is being poured into that, I'm sure it will just be a matter of time.

For development itself, both vscode and pycharm have ARM builds, so there's nothing blocking there.

**Comment by mariusandra** [comment]
Now that pandas and numpy [are gone](https://github.com/PostHog/posthog/pull/2997), PostHog should work on M1 macs. I'll give it a go later and report back. The only things needed should be:
- Install `grpcio` manually through pip
- Bump `cffi`

**Comment by mariusandra** [comment]
I got all packages to install with these extra steps:

1. `brew install openssl`
2. `GRPC_PYTHON_BUILD_SYSTEM_OPENSSL=1 GRPC_PYTHON_BUILD_SYSTEM_ZLIB=1 CFLAGS="-I/opt/homebrew/opt/openssl/include" LDFLAGS="-L/opt/homebrew/opt/openssl/lib" pip install -r requirements.txt`
3. Get an error like in [this comment](https://github.com/grpc/grpc/issues/25082#issuecomment-761298046), edit the `version.py` file like described there.
4. Run step 2 again
5. ???
6. Profit!

But, then I got this when running `migrate`:

```
django.core.exceptions.ImproperlyConfigured: Error loading psycopg2 module: dlopen(/Users/marius/Projects/PostHog/posthog/env/lib/python3.9/site-packages/psycopg2/_psycopg.cpython-39-darwin.so, 2): Symbol not found: _PQbackendPID
```

That's enough experimentation for today. To be continued.

**Comment by mariusandra** [comment]
The last error is fixed by replacing the intel based postgres.app with `brew install postgresql`. A shared library was shared between architectures and [was causing problems](https://github.com/psycopg/psycopg2/issues/1216). 

I fixed my borked setup by running:

```
python3 -m pip install --upgrade pip
SYSTEM_VERSION_COMPAT=1 pip install --force-reinstall psycopg2-binary
```

I also upgraded `psycopg2-binary` to 2.8.6 instead of 2.8.4, but not sure if that changed anything.

And with that posthog is running on a M1 mac!

![image](https://user-images.githubusercontent.com/53387/105643532-b5016080-5e90-11eb-9e99-b41c9852bb99.png)
